tests/test_simconfig.py

```diff
@@ -70,26 +70,23 @@ def test_init(matrices):
         ValueError, match="The standard deviation in amplitude"
     ):
         SimConfig(amp_sigma=-0.001)
+
+
+@pytest.mark.parametrize(
+    "noise_sample,",
+    [
+        ("dephasing", "depolarizing"),
+        ("eff_noise", "depolarizing"),
+        ("eff_noise", "dephasing"),
+        ("depolarizing", "eff_noise", "dephasing"),
+    ],
+)
+def test_init2(matrices, noise_sample):
     with pytest.raises(
         NotImplementedError,
         match="Depolarizing, dephasing and eff_noise channels",
     ):
-        SimConfig(noise=("dephasing", "depolarizing"))
-    with pytest.raises(
-        NotImplementedError,
-        match="Depolarizing, dephasing and eff_noise channels",
-    ):
-        SimConfig(noise=("eff_noise", "depolarizing"))
-    with pytest.raises(
-        NotImplementedError,
-        match="Depolarizing, dephasing and eff_noise channels",
-    ):
-        SimConfig(noise=("dephasing", "eff_noise"))
-    with pytest.raises(
-        NotImplementedError,
-        match="Depolarizing, dephasing and eff_noise channels",
-    ):
-        SimConfig(noise=("dephasing", "depolarizing", "eff_noise"))
+        SimConfig(noise=noise_sample)
```

Comment:
These should be isolated in a single test, otherwise you are repeating the rest of the function needlessly